src/GdbCommandHandler.cc

```diff
@@ -42,11 +42,16 @@ python
 import re
 
 def gdb_unescape(string):
+    len = len(string)
     result = ""
-    pos = 0
-    while pos < len(string):
-        result += chr(int(string[pos:pos+2], 16))
-        pos += 2
+    try:
+        pos = 0
+        while pos +2 <= len:
+            hex_char = string[pos:pos+2]
+            result += chr(int(hex_char, 16))
+            pos += 2
+    except: # check for unexpected string length and value
```

Comment:
Now you're just ignoring a trailing single character. So either this comment is wrong or you should be returning an empty string when the length is odd.

Reply:
will test and adjust later, the initial version was intended to throw, will check that part again...